src/compiler/factory.ts

```diff
@@ -2292,15 +2292,26 @@ namespace ts {
     /**
      * Sets a custom text range to use when emitting source maps.
      */
-    export function setSourceMapRange<T extends Node>(node: T, range: TextRange | undefined) {
+    export function setSourceMapRange<T extends Node>(node: T, range: SourceMapRange | undefined) {
         getOrCreateEmitNode(node).sourceMapRange = range;
         return node;
     }
 
     /**
+     * Create an external source map source file reference
+     */
+    export function createSourceMapSource(fileName: string, text: string, skipTrivia?: (pos: number) => number) {

```

Comment:
Please note that this method is also used in _src/harness/typeWriter.ts_. Please also note that while `SourceFileLike` and `SourceFileLike#getLineAndCharacterOfPosition` are public, the `computeLineAndCharacterOfPosition` function is internal.

I think my [comment above](https://github.com/Microsoft/TypeScript/pull/15935#discussion_r120485388) stands as the best approach at this point.